inst/rmarkdown/lua/latex-div.lua

```diff
@@ -4,15 +4,39 @@
      License: Public domain
 --]]
 
+-- COMMON PART ACROSS LUA FILTERS
+
+--[[
+  This function tests the pandoc version againt a target version
+  For Pandoc 2.7.3, PANDOC_VERSION >= "2.8" would be enough but before 2.7.3
+  it is a table object
+]]
+local function pandocAvailable(target)
+  -- this function only work for Pandoc 2.1 and above. It returns false is not
+  if not PANDOC_VERSION then return false end
+  if not target then error("No target version specified in pandocAvailable.") end
+  -- checking each version number
+  for i = 1,3 do
+    -- some versions do not have 3 numbers
+    if not PANDOC_VERSION[i] then break end
+    if target[i] and PANDOC_VERSION[i] < target[i] then
+      return false
```

Comment:
I think the logic here is wrong.
Let's say we have `target < PANDOC_VERSION` with

```lua
target = {2, 7, 3}
PANDOC_VERSION = {2, 8, 0}
```

The for loop will check all the index and the test becomes `true` at the third place.

1. `(2 and 2 < 2) -> false`
2. `(7 and 8 < 7) -> false`
3. `(3 and 0 < 3) -> true`

As a result, `pandocAvailable` returns `false`.

So I think below is what we need.

```lua
for i = 1,3 do
  if target[i] and target[i] == PANDOC_VERSION[i] then
    continue
  end
  if target[i] and target[i] < PANDOC_VERSION[i] then
    return true
  end
end

return false
```

Reply:
Thanks for the review. I did push a fix right after the PR though for the 4 versions number: cf4dde190cab1bae8ff5b5edaddce7ab2d9135a0

I'll modify for the case you described.

Reply:
See #2102